### PR TITLE
I2c

### PR DESCRIPTION
**Pull request by jamesadevine**
TO MERGE (after testing)

### CODE REVIEW

**Comment by jamesadevine** [comment]
@pelikhan @mmoskal

File: src/ZI2C.cpp [hunk 1]
```diff
@@ -0,0 +1,234 @@
...
+
+    DMESG("SDA pad %d, idx %d, fn: %d", sda_pin->sercom[0].pad, sda_pin->sercom[0].index, sda_fun);
+    DMESG("SCL pad %d, idx %d, fn: %d", scl_pin->sercom[0].pad, scl_pin->sercom[0].index, scl_fun);
+    DMESG("SERCOM IDX %d", sercomIdx);
+
+    gpio_set_pin_direction(sda.name, GPIO_DIRECTION_OUT);
+    gpio_set_pin_direction(scl.name, GPIO_DIRECTION_OUT);
+    gpio_set_pin_pull_mode(sda.name, GPIO_PULL_OFF);
+    gpio_set_pin_pull_mode(scl.name, GPIO_PULL_OFF);
+    gpio_set_pin_function(sda.name, PINMUX(sda.name, sda_fun));
```
[the remark this answers is not in the document]
Comment (jamesadevine) [reply src/ZI2C.cpp]: Ah, the reason i did it like this is because the constructor takes codal pins, not ZPins. Is it ok to specialise the constructor?

**Comment by mmoskal** [comment]
Yeah. No reason you would have any other kind of pin


________________________________

**Comment by jamesadevine** [comment]
Thanks @mmoskal :

https://github.com/lancaster-university/codal-samd/commit/299870f288c59bcdf8f9bba43ef229ac168e62a8

https://github.com/lancaster-university/codal-samd/commit/df13a57d44afb798018184de41fd66e2458c922b

**Pull request merged**